src/components/link/link.component.js

```diff
@@ -120,7 +97,11 @@ class InternalLink extends React.Component {
         hasContent={Boolean(children)}
         {...tagComponent("link", this.props)}
       >
-        {this.createLinkBasedOnType()}
+        {React.createElement(
```

Comment:
No need for `React.createElement` we can replace this with JSX now that the element doesn't change.